clang/lib/AST/Type.cpp

```diff
@@ -2400,7 +2402,66 @@ bool Type::isWebAssemblyTableType() const {
   return false;
 }
 
-bool Type::isSizelessType() const { return isSizelessBuiltinType(); }
+bool Type::isSizelessType() const {
```

Comment:
The name of this function sounds like this is a simple getter, but it's quite expensive. Are the other `Type::is*Type()` functions similarly complex or can we rename this?

Reply:
isIncompleteType has a recursive descent open to suggestions and/or renaming it though.